libraries/AP_OpticalFlow/AP_OpticalFlow.h

```diff
@@ -37,23 +37,21 @@
 
 class OpticalFlow_backend;
 
-class OpticalFlow
+class AP_OpticalFlow
 {
     friend class OpticalFlow_backend;
 
 public:
-    OpticalFlow();
+    AP_OpticalFlow();
 
-    /* Do not allow copies */
-    OpticalFlow(const OpticalFlow &other) = delete;
-    OpticalFlow &operator=(const OpticalFlow&) = delete;
+    CLASS_NO_COPY(AP_OpticalFlow);
 
     // get singleton instance
-    static OpticalFlow *get_singleton() {
+    static AP_OpticalFlow *get_singleton() {
         return _singleton;
     }
 
-    enum class OpticalFlowType {
+    enum class Type {
```

Comment:
I'm not a huge fan of using "Type" which is so close to the reserved word, "type".  Not a blocker perhaps but ..